include/osx/osd.h

```diff
@@ -46,27 +46,41 @@
 
 #include "unix/osd.h"
 
+#include "config.h"
+
+#if HAVE_CLOCK_GETTIME == 0
+
 #define CLOCK_REALTIME CALENDAR_CLOCK
 #define CLOCK_MONOTONIC SYSTEM_CLOCK
 
-#define pthread_yield pthread_yield_np
+typedef int clockid_t;
 
-#define bswap_64 OSSwapInt64
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int clock_gettime(clockid_t clk_id, struct timespec *tp);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
 
 #ifdef _POSIX_HOST_NAME_MAX
 #define HOST_NAME_MAX _POSIX_HOST_NAME_MAX
 #else
 #define HOST_NAME_MAX 255
 #endif
 
-typedef int clockid_t;
+#define pthread_yield pthread_yield_np
+
+#define bswap_64 OSSwapInt64
```

Comment:
What are the defines for `pthread_yield` and `bswap_64` for?  They don't look related to this `clock_gettime` fix...?